samza-api/src/main/java/org/apache/samza/operators/functions/SinkFunction.java

```diff
@@ -30,7 +31,7 @@
  */
 @InterfaceStability.Unstable
 @FunctionalInterface
-public interface SinkFunction<M>  extends InitableFunction, ClosableFunction {
+public interface SinkFunction<M>  extends InitableFunction, ClosableFunction, Serializable {
```

Comment:
We should document the serializability requirement for all these functions more explicitly.